plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/transport/OpenLineageHttpTransport.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage.transport;
+
+import com.google.inject.Inject;
+import io.openlineage.client.transports.HttpConfig;
+import io.openlineage.client.transports.HttpTransport;
+import io.openlineage.client.transports.TokenProvider;
+import io.trino.plugin.openlineage.config.OpenLineageClientHttpTransportConfig;
+
+import java.net.URI;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class OpenLineageHttpTransport
+        implements OpenLineageTransport
+{
+    private static class ApiKeyTokenProvider
+            implements TokenProvider
+    {
+        private final String token;
+
+        public ApiKeyTokenProvider(String token)
+        {
+            this.token = requireNonNull(token);
+        }
+
+        @Override
+        public String getToken()
+        {
+            return format("Bearer %s", this.token);
+        }
+    }
+
+    private final OpenLineageClientHttpTransportConfig config;
+
+    @Inject
+    public OpenLineageHttpTransport(OpenLineageClientHttpTransportConfig config)
+    {
+        this.config = config;
```

Comment:
Can we avoid maintaining config here - In general `Config` objects are mutable and they could potentially change in the future - So we could directly create some field for url and endpoint and update them accordingly or even initialize `HttpConfig` in the ctor.

Reply:
sure, created explicit attributes for this class